cli/src/utils.rs

```diff
@@ -41,6 +39,31 @@ pub fn format_devices(
     Ok(())
 }
 
+pub fn get_minimal_short_id_size<'a, I>(devices: I) -> usize
+where
+    I: Iterator<Item = &'a libparsec::DeviceID>,
+{
+    use itertools::Itertools;
+
+    devices
+        // Ensure we do not have duplicates in the set.
+        .sorted_by(|a, b| a.cmp(b))
+        .dedup_by(|a, b| a == b)
+        // Generate combinations of the ids
+        .tuple_combinations::<(_, _)>()
+        // Find the first different bit between the two ids:
+        // We use the fact that we will display the id in hex format to compare both ids using the bitwise XOR and count the leading zeros.
+        // Since it takes 4 bits to represent a hex digit, we divide the leading zeros by 4 to get the position of the first different hex digit.
+        .map(|(a, b)| {
+            let cmp = a.as_u128() ^ b.as_u128();
+            let position = cmp.leading_zeros() / 4;
+            position as usize + 1
+        })
+        // Find the max value or fallback to 1 (meaning we have only one device)
+        .max()
+        .unwrap_or(1)
```

Comment:
I would keep a lower bound on three characters (that's what we did on Parsec v2), given having a single character as ID makes it harder to understand it is an ID, and harder to copy&paste (double click on a word to select it is much simpler than selecting a single character)

Reply:
done